src/main/java/kitchenpos/products/application/ProductService.java

```diff
@@ -32,50 +38,41 @@ public ProductService(
     }
 
     @Transactional
-    public Product create(final Product request) {
-        final BigDecimal price = request.getPrice();
-        if (Objects.isNull(price) || price.compareTo(BigDecimal.ZERO) < 0) {
-            throw new IllegalArgumentException();
-        }
-        final String name = request.getName();
-        if (Objects.isNull(name) || purgomalumClient.containsProfanity(name)) {
-            throw new IllegalArgumentException();
-        }
-        final Product product = new Product();
-        product.setId(UUID.randomUUID());
-        product.setName(name);
-        product.setPrice(price);
-        return productRepository.save(product);
+    public CreateProductResponseDto create(final CreateProductRequestDto request) {
+        final DisplayedName displayedName = new DisplayedName(request.getName(), purgomalumClient);
+        final ProductPrice productPrice = new ProductPrice(request.getPrice());
+        final Product product = new Product(UUID.randomUUID(), displayedName, productPrice);
+        return CreateProductResponseDto.from(productRepository.save(product));
     }
 
     @Transactional
-    public Product changePrice(final UUID productId, final Product request) {
-        final BigDecimal price = request.getPrice();
-        if (Objects.isNull(price) || price.compareTo(BigDecimal.ZERO) < 0) {
-            throw new IllegalArgumentException();
-        }
+    public ChangeProductPriceResponseDto changePrice(final UUID productId,
+        final ChangeProductPriceRequestDto request) {
         final Product product = productRepository.findById(productId)
             .orElseThrow(NoSuchElementException::new);
-        product.setPrice(price);
+        product.changePrice(request.toValueObject());
```

Comment:
```suggestion
        product.changePrice(request.getPrice());
```
위와 같은 형태로 changePrice메서드를 호출할 수 있게 
request DTO의 값객체 변환 메서드의 처리로직을 changePrice메서드 내부로 옮겨 보는 건 어떻게 생각하세요?

Reply:
제가 toValueObject() 메소드를 통해 DTO -> VO 변환했던 이유는 아래와 같은데, 학성님의 의도도 충분히 이해되어 코드 반영하였습니다!

DTO 내부에서 변환 메소드를 제공했던 이유
- 변환 책임이 DTO에 있는 경우 직관적이라고 생각
- 해당 메소드명만 보고도 DTO를 VO로 변환하는 역할이라고 유추 가능

학성님께서 VO 변환 로직을 changePrice() 내부로 옮기는 것을 제안하신 이유는 아래와 같다고 생각합니다.

- DTO는 단순히 데이터 전달 역할만 수행해야 한다는 원칙

의도를 제대로 이해한 게 맞을까요? 혹시 더 설명해주실 부분 있다면 보충해주시면 감사하겠습니다 :)

Reply:
> 제가 toValueObject() 메소드를 통해 DTO -> VO 변환했던 이유는 아래와 같은데, 학성님의 의도도 충분히 이해되어 코드 반영하였습니다!
> 
> DTO 내부에서 변환 메소드를 제공했던 이유
> 
> * 변환 책임이 DTO에 있는 경우 직관적이라고 생각
> * 해당 메소드명만 보고도 DTO를 VO로 변환하는 역할이라고 유추 가능
> 
> 학성님께서 VO 변환 로직을 changePrice() 내부로 옮기는 것을 제안하신 이유는 아래와 같다고 생각합니다.
> 
> * DTO는 단순히 데이터 전달 역할만 수행해야 한다는 원칙
> 
> 의도를 제대로 이해한 게 맞을까요? 혹시 더 설명해주실 부분 있다면 보충해주시면 감사하겠습니다 :)

네 도원님의 생각도 충분히 일리있는 의견 같아요!
다만 제가 위 코드를 제안했던 이유는 다음과 같습니다.

1. toVAlueObject라는 메서드명은 VO로 변환한다는 의미는 즉시 파악되지만 어떤 항목을 변환하는지가 모호해 보였습니다.
    예를들어 Request DTO에 요구사항에 따라 price뿐만아니라 quantity같은 다른 필드도 추가될 수 있기 때문입니다.

2. ProductPrice는 도메인 계층에서 다루는 VO이므로 DTO보다는 도메인 내부에서 생성하는 것이 자연스럽다고 생각했습니다.
    DTO가 VO와 강하게 결합되면 DTO가 변경될 때 VO에도 영향이 갈 수도 있기 때문입니다.

3. DTO의 단일책임은 이미 파악하신 것 같네요

상황에 따라서 변환로직을 도메인 객체나 DTO가 아닌 별도의 변환 클래스(mapper)로 분리하여 애플리케이션 계층에 위치시키는 방법도 있을 것 같습니다.